src/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs

```diff
@@ -28,48 +28,86 @@
+    /// Options object with string values loadable from a configuration file 
+    /// (for instance a JSon file as in an asp.net configuration scenario)
+    /// See https://aka.ms/msal-net-application-configuration
     /// </summary>
     public abstract class ApplicationOptions
     {
         /// <summary>
+        /// Client ID (also known as App ID) of the application as registered in the
+        /// application registration portal (https://aka.ms/msal-net-register-app)
         /// </summary>
         public string ClientId { get; set; }
 
         /// <summary>
-        /// Can be domain, can be guid tenant, can be meta-tenant (e.g. consumers).
+        /// Tenant from which the application will allow users to sign it. This can be:
+        /// a domain associated with a tenant, a guid (tenant id), or a meta-tenant (e.g. consumers).
+        /// This property is mutually exclusive with <see cref="AadAuthorityAudience"/>
         /// </summary>
+        /// <remarks>The name of the property was chosen o ensure compatibility with AzureAdOptions 
+        /// in ASP.NET Core configuration files (even the semantics would be tenant)</remarks>
         public string TenantId { get; set; }
 
         /// <summary>
-        /// Mutually exclusive with TenantId...
+        /// Sign-in audience. This property is mutually exclusive with TenantId...
         /// </summary>
         public AadAuthorityAudience AadAuthorityAudience { get; set; } = AadAuthorityAudience.None;
 
         /// <summary>
-        /// For compat with AzureAdOptions...
+        /// STS instance (for instance https://login.microsoftonline.com for the Azure public cloud). 
+        /// The name was chosen to ensure compatibility with AzureAdOptions in ASP.NET Core.
+        /// This property is mutually exclusive with <see cref="AzureCloudInstance"/>
         /// </summary>
         public string Instance { get; set; }
 
         /// <summary>
-        /// Mutually exclusive with Instance, allows users to use the enum instead of the explicit url.
+        /// Specific instance in the case of Azure Active Directory. 
+        /// This property is mutually exclusive with <see cref="Instance"/>. 
+        /// It allows users to use the enum instead of the explicit url.
         /// </summary>
         public AzureCloudInstance AzureCloudInstance { get; set; } = AzureCloudInstance.None;
 
         /// <summary>
+        /// The redirect URI (also known as Reply URI or Reply URL), is the URI at which Azure AD will contact back the application with the tokens.
+        /// This redirect URI needs to be registered in the app registration (https://aka.ms/msal-net-register-app).
+        /// In MSAL.NET, <see cref="T:IPublicClientApplication"/> define the following default RedirectUri values:
+        /// <list type="bullet">
+        /// <item><description><c>urn:ietf:wg:oauth:2.0:oob</c> for desktop (.NET Framework and .NET Core) applications</description></item>
```

Comment:
Don't we expose a constant for this uri?

Reply:
If a redirecturi isn't specified here then we will fall back to a default based on the defaulturi value in the platformproxy.

---
In reply to: [248327366](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/790#discussion_r248327366) [](ancestors = 248327366)